src/draw/sw/lv_draw_sw_arc.c

```diff
@@ -126,9 +126,9 @@ void lv_draw_sw_arc(lv_draw_unit_t * draw_unit, const lv_draw_arc_dsc_t * dsc, c
         lv_image_decoder_open(&decoder_dsc, dsc->img_src, NULL);
         img_area.x1 = 0;
         img_area.y1 = 0;
-        img_area.x2 = decoder_dsc.header.w - 1;
-        img_area.y2 = decoder_dsc.header.h - 1;
-        int32_t ofs = decoder_dsc.header.w / 2;
+        img_area.x2 = decoder_dsc.decoded->header.w - 1;
```

Comment:
Is it really correct in every case?
I mean: why do we have the `lv_image_header_t` struct duplication then in [`_lv_image_decoder_dsc_t`](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_image_decoder.h#L146) and in it's [`lv_draw_buf_t * decoded`](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_image_decoder.h#L150) ?

Reply:
The decoded image header can be different in stride, color format, or flags.